03_letters/lexual/freq.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func letters(s string) map[rune]int {
+	m := make(map[rune]int)
+	for _, r := range s {
+		m[r]++
+	}
+	return m
+}
+
+func sortLetters(m map[rune]int) []string {
+	keys := make([]rune, len(m))
+	i := 0
+	for k := range m {
+		keys[i] = k
+		i++
+	}
+	// sort keys
+	sort.Slice(keys, func(i, j int) bool {
+		return keys[i] < keys[j]
+	})
+	sorted := make([]string, len(m))
+	for i, k := range keys {
```

Comment:
I believe sorting should be done by `value`(occurrences) not by the `key`.

Reply:
The readme example seems to be sorted by key:

https://github.com/anz-bank/go-course/blob/master/03_letters/README.md

Reply:
I just checked merged examples and they are done in the both ways, so it's fine.